src/reducers/moveThought.ts

```diff
@@ -158,39 +41,118 @@ const moveThought = (state: State, { oldPath, newPath, offset, skipRerank, newRa
 
   const newCursorPath = isPathInCursor ? (isCursorAtOldPath ? newPath : getUpdateCursor()) : state.cursor
 
+  const sourceThoughtPath = oldPath
+  const destinationThoughtPath = rootedParentOf(state, newPath)
+  const sourceThoughtId = head(sourceThoughtPath)
+  const destinationThoughtId = head(destinationThoughtPath)
+
+  const sourceThought = getThoughtById(state, sourceThoughtId)
+  const sourceParentThought = getThoughtById(state, sourceThought.parentId)
+  const destinationThought = getThoughtById(state, destinationThoughtId)
+
+  const sameContext = sourceParentThought.id === destinationThoughtId
+  const childrenOfDestination = getChildrenRankedById(state, destinationThoughtId)
+
+  /**
+   * Find duplicate thought.
+   */
+  const duplicateSubthought = () =>
+    childrenOfDestination.find(child => normalizeThought(child.value) === normalizeThought(sourceThought.value))
+
+  // if thought is being moved to the same context that is not a duplicate case
+  const duplicateThought = !sameContext ? duplicateSubthought() : undefined
+
+  const isPendingMerge = duplicateThought && (sourceThought.pending || duplicateThought.pending)
+
+  const newStateAfterMerge =
+    !!duplicateThought &&
+    !isPendingMerge &&
+    mergeThoughts(state, {
+      sourceThoughtPath,
+      targetThoughtPath: appendToPath(destinationThoughtPath, duplicateThought.id),
+    })
+
+  const destinationContext = pathToContext(state, destinationThoughtPath)
+
+  const isArchived = destinationContext?.indexOf('=archive') !== -1
+
+  // if move is used for archive then update the archived field to latest timestamp
+  const archived = isArchived ? timestamp() : destinationThought.archived
+
+  const contextIndexUpdates = newStateAfterMerge
+    ? {}
+    : {
```

Comment:
Can you add a comment explaining what the updates are and why they are only needed if `newStateAfterMerge` is null?

Reply:
Sure.